### PR TITLE
Add Log Level Options to Project Manager

### LINKED ISSUE #945: Add Trace-Level Logging to the Application Initialization Logic
### Summary

There is a case when the application hangs during the startup. We should be able to debug the initialization process by adding some low-level logging to the initialization logic.

### Value

Be able to debug cases when the component fails during the application startup.

### Specification

- [ ] Be able to choose log level (the level of verbosity) when starting the application (something like `-vvv`)
- [ ] Add trace logging to the project manager startup logic
- [ ] Add trace logging to the language server startup logic

### Acceptance Criteria & Test Cases

Application can be started with `-vvv` (very very verbose) log level and prints the initialization logs.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #945 

Set the project manager log level from the CLI options. Now the logs verbosity can be controlled by stacking `-v` verbosity options:
- `<no options>` - _info_
- `-v` - _debug_
- `-vv` - _trace_

Changelog:
- add CLI options parser to project manager
- update project manager to change the application log level based on CLI verbosity options

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: lib/scala/project-manager/src/main/scala/org/enso/projectmanager/boot/Cli.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      .builder("h")
+      .longOpt(HELP_OPTION)
+      .desc("Displays this message.")
+      .build()
+
+    val verbose: cli.Option = cli.Option
+      .builder("v")
+      .longOpt(VERBOSE_OPTION)
+      .desc("Increase logs verbosity.")
+      .build()
```
Comment (radeusgd) [inline lib/scala/project-manager/src/main/scala/org/enso/projectmanager/boot/Cli.scala]: I had the impression that we shouldn't use short option names (but maybe this is an exception? definitely writing `--verbose --verbose` wouldn't be great).

If this option can indeed use the short name, I suggest the help text is made a bit more explanatory - what is the default verbosity level? Also from the current help text I'm not sure if I would guess that I can add `-v` twice to get to an even higher level (I know this is pretty common in many programs, but I think it should be explained).

File: lib/scala/project-manager/src/main/scala/org/enso/projectmanager/util/Logging.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  private val ROOT_LOGGER = "org.enso"
+
+  /**
+    * Set log level for the application root logger.
+    *
+    * @param level the log level
+    * @return the new log level
+    */
+  def setLogLevel(level: LogLevel): Option[LogLevel] = {
```
Comment (radeusgd) [inline lib/scala/project-manager/src/main/scala/org/enso/projectmanager/util/Logging.scala]: What is the rationale of returning `Option` here and in the getter? It seems just returning the `Try` would carry more information (in the callsite, if setting log level fails, the application can report what was the cause of failure).

**Review by radeusgd — APPROVED** [review]
Looks very good, just two small issues I'd like to discuss.

File: lib/scala/project-manager/src/main/scala/org/enso/projectmanager/boot/Cli.scala (hunk 1 above)
Comment (4e6) [reply lib/scala/project-manager/src/main/scala/org/enso/projectmanager/boot/Cli.scala]: I think `-v` is fine for verbosity. I'll add extra info about options stacking :+1:

File: lib/scala/project-manager/src/main/scala/org/enso/projectmanager/util/Logging.scala (hunk 2 above)
Comment (4e6) [reply lib/scala/project-manager/src/main/scala/org/enso/projectmanager/util/Logging.scala]: I'll check what is the error message reported. If it is readable, I'll print it as well :+1:

File: lib/scala/project-manager/src/main/scala/org/enso/projectmanager/util/Logging.scala [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  /** Get log level of the application root logger. */
+  def getLogLevel: Either[Throwable, LogLevel] = {
+    Try {
+      val ctx   = LoggerFactory.getILoggerFactory.asInstanceOf[LoggerContext]
+      val level = ctx.getLogger(ROOT_LOGGER).getLevel
+      LogLevel.fromLogback(level)
+    }.toEither
+  }
```
Comment (lolczak) [inline lib/scala/project-manager/src/main/scala/org/enso/projectmanager/util/Logging.scala]: Please use `Either.catchNonFatal` from cats lib.

**Review by lolczak — APPROVED** [review]
LGTM

**Pull request merged**